tests/unit/test_search_model.py

```diff
@@ -293,6 +293,17 @@ def test_index_search_request():
         '"requestMetadata": {"saveSearchLog": false, "utmTags": ["project_sdk_python"]}}'
     )
 
+    # deserialization test
+    data = {
+        "query": {"term": {"__typeName.keyword": {"value": "Schema"}}},
+        "from": 0,
+        "size": 300,
+        "track_total_hits": True,
+        "sort": [],
+    }
+    test_dsl = DSL(**data)
+    assert test_dsl.query == data["query"]  # TODO: <-- make sure its Query object
```

Comment:
```py
class DSL(AtlanObject):
    from_: int = Field(default=0, alias="from")
    size: int = Field(default=300)
    aggregations: Dict[str, Aggregation] = Field(default_factory=dict)
    track_total_hits: Optional[bool] = Field(default=True, alias="track_total_hits")
    post_filter: Optional[Query] = Field(default=None, alias="post_filter")
    query: Optional[Query]
    req_class_name: Optional[str] = Field(default=None, exclude=True)
    sort: List[SortItem] = Field(default_factory=list, alias="sort")
```

This sample test was written to test the logic; it's not complete. Please move it to a separate unit test case and ensure it includes a complex (nested) data dictionary that covers all DSL model fields ^^ 🙏 
 
```suggestion
    data = {
        "query": {"term": {"__typeName.keyword": {"value": "Schema"}}},
        "from": 0,
        "size": 300,
        "track_total_hits": True,
        "sort": [],
    }
    test_dsl = DSL(**data)
    assert test_dsl.query == data["query"] 
```

Reply:
Okay, since we already have `test_dsl_serialization_and_deserialization`, please remove the above test. ^^

Reply:
Got it